enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,640 @@
+type VSpherePlatformFailureDomainHosts struct {
+	// VMGroupName is the name of the VM group
+	VMGroupName string `json:"vmGroupName"`
+
+	// HostGroupName is the name of the Host group
+	HostGroupName string `json:"hostGroupName"`
+}
+
+```
+```golang
+// VSphereFailureDomainSpec defines the desired state of VSphereFailureDomain
+type VSpherePlatformFailureDomainSpec struct {
+
+	// Region defines the name and type of a region
+	Region VSpherePlatformFailureDomain `json:"region"`
+
+	// Zone defines the name and type of a zone
+	Zone VSpherePlatformFailureDomain `json:"zone"`
+
+	// Topology describes a given failure domain using vSphere constructs
+	Topology VSpherePlatformTopology `json:"topology"`
+}
+
+```
+```golang
+type VSpherePlatformFailureDomain struct {
+	// Name is the name of the tag that represents this failure domain
+	Name string `json:"name"`
+
+	// Type is the type of failure domain, the current values are "Datacenter", "ComputeCluster" and "HostGroup"
+	// +kubebuilder:validation:Enum=Datacenter;ComputeCluster;HostGroup
+	Type VSpherePlatformFailureDomainType `json:"type"`
+
+	// TagCategory is the category used for the tag
+	TagCategory string `json:"tagCategory"`
+
+    
+    // ** hmm ** thinking no on this... ?
+    
+	// AutoConfigure tags the Type which is specified in the Topology
+	//AutoConfigure *bool `json:"autoConfigure,omitempty"`
+}
+```
+```golang
+type VSpherePlatformTopology struct {
+	// Datacenter as the failure domain.
+	// +kubebuilder:validation:Required
+	Datacenter string `json:"datacenter"`
+
+	// ComputeCluster as the failure domain
+	// +optional
+	ComputeCluster *string `json:"computeCluster,omitempty"`
+
+    
+	// Hosts has information required for placement of machines on VSphere hosts.
+	// +optional
+	Hosts *VSpherePlatformFailureDomainHosts `json:"hosts,omitempty"`
+    
+
+	// Networks is the list of networks within this failure domain
+	// +optional
+	Networks []string `json:"networks,omitempty"`
+
+	// Datastore is the name or inventory path of the datastore in which the
+	// virtual machine is created/located.
+	// +optional
+	Datastore string `json:"datastore,omitempty"`
+
+	// ResourcePool is the absolute path of the resource pool where virtual machines will be
+	// created. The absolute path is of the form /<datacenter>/host/<cluster>/Resources/<resourcepool>.
+	// +optional
+	ResourcePool string `json:"resourcepool,omitempty"`
+}
+```
+```golang
+// VCenter stores the vCenter connection fields and the Region slice
+// that is used to create virtual machines in various datacenters and clusters
+type VSpherePlatformVCenterSpec struct {
+
+	// Server is the domain name or IP address of the vCenter.
+	// +required
+	Server string `json:"server"`
+
+	// User is the name of the user to use to connect to the vCenter.
+	// +required
+	User string `json:"user"`
+
+	// Port is the vCenter API TCP port that the vCenter SDK clients connect to.
+	// +optional
+	Port uint `json:"port,omitempty"`
```

Comment:
I'm not sure about unsigned vs not, but this should be a sized int for sure, either, I suspect an `int32` or `uint32`

Reply:
I think this is where I got it from
https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/common/config/types_yaml.go#L90